### PR TITLE
Add binder links to rendered notebooks

### LINKED ISSUE #405: Add binder links to mkdocs
These were introduced in 3c744892c3717eda709d017c07954d9b0a3de0ce for sphinx

### PR DESCRIPTION
**Pull request by AnesBenmerzoug**
### Description

This PR closes #405 

image

### Changes

- Uses `mknotebooks` instead of `mkdocs-jupyter`.
- Added hook to replace binder url created by mknotebooks.

### Checklist

- [ ] Wrote Unit tests (if necessary)
- [ ] Updated Documentation (if necessary)
- [ ] Updated Changelog
- [ ] If notebooks were added/changed, added boilerplate cells are tagged with `"nbsphinx":"hidden"`

### CODE REVIEW

File: docs/css/extra.css [hunk 1]
```diff
@@ -109,3 +109,18 @@ body[data-md-color-scheme="default"] .invertible img {
 body[data-md-color-scheme="slate"] .invertible img {
     filter: invert(100%) hue-rotate(180deg);
 }
+
+/* Rendered dataframe from jupyter */
+div.rendered_html {
+    width: 100% !important;
+    overflow-x: scroll;
```
Comment (mdbenito) [inline docs/css/extra.css:116]: This creates scrollbars all over the place on my system:
image

Removing it seems not to cause any problem. Is it necessary for you?

**Review by mdbenito — COMMENTED** [review]
In this branch I see tons of warnings from mkdocs that weren't there before, like the following:

```
WARNING -  Doc file '_build/html/examples/shapley_utility_learning.ipynb' contains a relative link '../pydvl/value/shapley/common.rst#pydvl.value.shapley.common.compute_shapley_values', but the target '_build/html/pydvl/value/shapley/common.rst' is not found among documentation files.
```
Any idea why?'

Also: the launch binder button is enough, the "binder" caption / subtitle is redundant and quite ugly. I would limit the whole thing to the anchor tag:

image

File: docs/css/extra.css (hunk 1 above)
Comment (AnesBenmerzoug) [reply docs/css/extra.css]: Done.

**Comment by AnesBenmerzoug** [comment]
@mdbenito Unlike [`mkdocs-jupyter`](https://github.com/danielfrg/mkdocs-jupyter/blob/main/mkdocs_jupyter/plugin.py#L107), [`mknotebooks`](https://github.com/greenape/mknotebooks/blob/master/mknotebooks/plugin.py#L263) converts the notebooks to markdown and lets mkdocs handle the conversion from markdown to HTML. That makes it work with other plugins and extensions. That's why those warnings started appearing. We could now convert all the links to the jupyter notebooks to use aliases and let mkdocs resolve them but that makes the links unreadable in the notebook itself.

**Comment by AnesBenmerzoug** [comment]
I removed the Binder caption.

**Comment by mdbenito** [comment]
> @mdbenito Unlike [`mkdocs-jupyter`](https://github.com/danielfrg/mkdocs-jupyter/blob/main/mkdocs_jupyter/plugin.py#L107), [`mknotebooks`](https://github.com/greenape/mknotebooks/blob/master/mknotebooks/plugin.py#L263) converts the notebooks to markdown and lets mkdocs handle the conversion from markdown to HTML. That makes it work with other plugins and extensions. That's why those warnings started appearing. We could now convert all the links to the jupyter notebooks to use aliases and let mkdocs resolve them but that makes the links unreadable in the notebook itself.

I see. But the links are already broken in the notebooks since they are relative to the built docs, right? It think we could convert them.

**Pull request merged**